consensus/src/experimental/buffer_manager.rs

```diff
@@ -254,6 +184,112 @@ impl StateManager {
         Ok(())
     }
 
+    /// this function updates the buffer according to the response from the execution phase
+    /// it also initiates a request to the signing phase.
+    async fn update_buffer_execution_resp(
+        &mut self,
+        execution_resp: ExecutionResponse,
+    ) -> anyhow::Result<()> {
+        let ExecutionResponse { inner } = execution_resp;
+
+        let executed_blocks = inner?;
+
+        let current_cursor = get_next(&self.execution_root); // must be Some
```

Comment:
is `must be Some` true? what if the root is already the tail?

Reply:
Previously I handled the empty list case in the outer function but later removed that. I've added a fix.